### ISSUE TITLE
fix add semi-colon error

### ISSUE THREAD

**Issue by WingDust** [sqlfluff/sqlfluff#7362]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

```sql
-- cat test.sql
select string_agg(content, chr(10))
from node

select string_agg(content, chr(10))
from node
-- cat .sqlfluff
[sqlfluff]
dialect = postgres
exclude_rules = LT05, LT14
rules = convention.terminator

[sqlfluff:indentation]
tab_space_size = 2

[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = lower

[sqlfluff:rules:convention.terminator]
require_final_semicolon = True
multiline_newline = True
-- stdout: sqlfluff fix test.sql
==== finding fixable violations ====
== [test.sql] FAIL
L:   5 | P:  10 | CV06 | Statements must end with a semi-colon.
                       | [convention.terminator]
  [1 templating/parsing errors found]
==== no fixable linting violations found ====
All Finished!
  [1 unfixable linting violations found]
```
not fix add semi-colon

### Expected Behaviour

select string_agg(content, chr(10))
from node;

select string_agg(content, chr(10))
from node;

### Observed Behaviour

select string_agg(content, chr(10))
from node

select string_agg(content, chr(10))
from node

### How to reproduce

![Image](https://github.com/user-attachments/assets/0b9d9eec-106b-4f51-98c9-5e2ad4801c7d)```sql
-- cat test.sql
select string_agg(content, chr(10))
from node

select string_agg(content, chr(10))
from node
-- cat .sqlfluff
[sqlfluff]
dialect = postgres
exclude_rules = LT05, LT14
rules = convention.terminator

[sqlfluff:indentation]
tab_space_size = 2

[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = lower

[sqlfluff:rules:convention.terminator]
require_final_semicolon = True
multiline_newline = True
-- stdout: sqlfluff fix test.sql
==== finding fixable violations ====
== [test.sql] FAIL
L:   5 | P:  10 | CV06 | Statements must end with a semi-colon.
                       | [convention.terminator]
  [1 templating/parsing errors found]
==== no fixable linting violations found ====
All Finished!
  [1 unfixable linting violations found]
```

### Dialect

postgres

### Version

3.5.0

### Configuration

[sqlfluff]
dialect = postgres
exclude_rules = LT05, LT14
rules = convention.terminator

[sqlfluff:indentation]
tab_space_size = 2

[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = lower

[sqlfluff:rules:convention.terminator]
require_final_semicolon = True
multiline_newline = True


### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by peterbud** [comment]
Hi @WingDust, thanks for submitting the issue.

In Postgres AFAIK the statement terminator is mandatory in multi-statement scripts. That means that the above example is syntactically incorrect, therefore it is unparsable after the first statement. The `fix` command can only fix linting errors, but it cannot fix syntax errors.

**Comment by peterbud** [comment]
Closing for now. If there is a response we can re-open
